tests/conservation_sum_rules/__main__.py

```diff
@@ -0,0 +1,69 @@
+"""
+Compute and save collision integrals of active neutrinos
+in the case of elastic processes only.
+"""
+
+import os
+import numpy as np
+
+from particles import Particle
+from library.SM import particles as SMP, interactions as SMI
+from evolution import Universe
+from common import UNITS, Params
+from common import LinearSpacedGrid
+
+folder = os.path.join(os.path.split(__file__)[0], 'output')
+
+T_final = 0.03 * UNITS.MeV
+params = Params(T=10 * UNITS.MeV, dy=0.003125)
+
+universe = Universe(params=params, folder=folder)
+
+linear_grid = LinearSpacedGrid(MOMENTUM_SAMPLES=51, MAX_MOMENTUM=20*UNITS.MeV)
+
+photon = Particle(**SMP.photon)
+electron = Particle(**SMP.leptons.electron)
+neutrino_e = Particle(**SMP.leptons.neutrino_e, **{'grid': linear_grid})
+neutrino_mu = Particle(**SMP.leptons.neutrino_mu, **{'grid':linear_grid})
+neutrino_tau = Particle(**SMP.leptons.neutrino_tau,  **{'grid': linear_grid})
+
+neutrino_e.decoupling_temperature=9.5 * UNITS.MeV
+neutrino_mu.decoupling_temperature=9.5 * UNITS.MeV
+neutrino_tau.decoupling_temperature=9.5 * UNITS.MeV
+
+universe.add_particles([photon, electron, neutrino_e, neutrino_mu, neutrino_tau])
+
+interactions_pre = (
+	SMI.neutrino_interactions(leptons=[electron], neutrinos=[neutrino_e, neutrino_mu, neutrino_tau])
+)
+
+# for-loop below gets rid of non-elastic processes.
+# For some reason operators as del/remove/pop don't work properly here. This is the only way I managed to make it work.
+# And yes, I know, it looks terrible.
+for x in interactions_pre:
+    for n, y in enumerate(x.integrals):        
+        if (y.reaction[0].specie.symbol != y.reaction[2].specie.symbol and y.reaction[0].specie.symbol != y.reaction[3].specie.symbol): 
```

Comment:
Your check is not very good. Indeed, this is how you can roughly filter some of the integrals you don't want to see, but you missed the following:

Every reaction creates multiple integrals. `A + B -> C + D` creates 4 terms in 4 different Boltzmann equations. `A + B -> A + B` will produce 2: `I_A(A+B->A+B)` and `I_B(B+A->A+B)`. The general notation is that the first particle in the integral is the one corresponding to Boltzmann equations. Others are ordered by species for convenience.

I would make a check like this:

```
from collections import Counter

left_species = Counter(item.specie for item in integral.reaction if item.side == -1)
right_species = Counter(item.specie for item in integral.reaction if item.side == 1)

if left_species == right_species:
```

Also, one-letter variable names produce lots of bugs. Python is designed such that you really can program without counters `i, j, k, l, ...` familiar from C. So there is no reason to call an `interaction` `x`